com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -541,23 +541,44 @@ internal static void ServerResetShudownStateForSceneObjects()
             }
         }
 
+        /// <summary>
+        /// Gets called only by NetworkSceneManager.SwitchScene
+        /// </summary>
         internal static void ServerDestroySpawnedSceneObjects()
         {
+            //This Allocation is "ok" for now because this code only executes when a new scene is switched to
+            //We need to keep track of GameObjects and NetworkObjects to be destroyed in the current scene
+            //but only destroy them once outside of the enumerative foreach loop
```

Comment:
Ok...would you also capture the 'why' here too in the comment?

Reply:
Simplified the code required and adjusted the comment.